pkg/parse/parse.go

```diff
@@ -231,79 +239,45 @@ func validateIPAddress(val string) (string, error) {
 	return "", fmt.Errorf("%s is not an ip address", val)
 }
 
-// ValidateFlags searches for StringFlags or StringSlice flags that never had
-// a value set.  This commonly occurs when the CLI mistakenly takes the next
-// option and uses it as a value.
-func ValidateFlags(c *cli.Context, flags []cli.Flag) error {
-	re, err := regexp.Compile("^-.+")
-	if err != nil {
-		return errors.Wrap(err, "compiling regex failed")
-	}
-
-	// The --cmd flag can have a following command i.e. --cmd="--help".
-	// Let's skip this check just for the --cmd flag.
-	for _, flag := range flags {
-		switch reflect.TypeOf(flag).String() {
-		case "cli.StringSliceFlag":
-			{
-				f := flag.(cli.StringSliceFlag)
-				name := strings.Split(f.Name, ",")
-				if f.Name == "cmd" {
-					continue
-				}
-				val := c.StringSlice(name[0])
-				for _, v := range val {
-					if ok := re.MatchString(v); ok {
-						return errors.Errorf("option --%s requires a value", name[0])
-					}
-				}
-			}
-		case "cli.StringFlag":
-			{
-				f := flag.(cli.StringFlag)
-				name := strings.Split(f.Name, ",")
-				if f.Name == "cmd" {
-					continue
-				}
-				val := c.String(name[0])
-				if ok := re.MatchString(val); ok {
-					return errors.Errorf("option --%s requires a value", name[0])
-				}
-			}
-		}
-	}
-	return nil
-}
-
```

Comment:
FYI @baude and I talked, Cobra does this for us, so this ValidateFlags function is no longer needed.